crypto/ocsp/ocsp_vfy.c

```diff
@@ -31,6 +31,7 @@ static int ocsp_verify_signer(X509 *signer, X509_STORE *st, unsigned long flags,
 {
     X509_STORE_CTX *ctx = X509_STORE_CTX_new();
     X509_VERIFY_PARAM *vp;
+    int verifying_reponse = chain != NULL;
```

Comment:
This seems to assume that if there is no chain that it's a response, which doesn't seem to be the case.

Reply:
See the two uses of `ocsp_verify_signer()`: one is for responses with a non-NULL parameter,
and the other one is for verifying requests there the parameter is NULL.

Reply:
When reviewing code please make sure to view and understand it in context, 
also for your other comment above.

Reply:
I actually looked at that function, but parsed it wrong. For some reason, if OCSP_NOCHAIN doesn't have an effect on the chain that's passed, but on untrusted that's passed, and chain is always NULL.

In any case, I find this a bad style, it makes assumptions that some day might no longer hold.

Reply:
Clearly I need to look better, chain is what's returned.

Reply:
So this is resolved?

Reply:
I still think this is bad style.

Reply:
All right, added an extra parameter to `ocsp_verify_signer()` fixing this style issue.